### ISSUE TITLE
SA1119 StatementMustNotUseUnnecessaryParenthesis incorrectly reported with await

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#388]
The following expression incorrectly results in SA1119.

```csharp
(await document.GetSyntaxRootAsync().ConfigureAwait(false)).ToFullString()
```

**Comment by sharwell** [comment]
@pdelvo I assigned this to you since you implemented the initial diagnostic.
